packages/grafana-data/src/transformations/transformers/calculateField.ts

```diff
@@ -315,22 +315,22 @@ function getCenteredWindowValues(frame: DataFrame, reducer: ReducerID, selectedF
       if (i === 0) {
         // We're at the start and need to prime the leading part of the window
         for (let x = 0; x < leadingPartOfWindow + 1 && x < selectedField.values.length; x++) {
-          if (selectedField.values[x] !== null) {
+          if (selectedField.values[x] !== null && selectedField.values[x] !== undefined) {
             sum += selectedField.values[x];
             count++;
           }
         }
       } else {
         if (last < selectedField.values.length) {
           // Last is inside the data and should be added.
-          if (selectedField.values[last] !== null) {
+          if (selectedField.values[last] !== null && selectedField.values[last] !== undefined) {
             sum += selectedField.values[last];
             count++;
           }
         }
         if (first > 0) {
           // Remove values that have fallen outside of the window, if the start of the window isn't outside of the data.
-          if (selectedField.values[first - 1] !== null) {
+          if (selectedField.values[first - 1] !== null && selectedField.values[first] !== undefined) {
```

Comment:
first - 1?

Reply:
Oh my, that was sloppy of me.

Should have added some tests for the undefined case from the start... Anyway, they're added now.